### PR TITLE
[Settings v2] Homepage and manage identites

### PR DESCRIPTION
**Pull request by IniZio**
#### Settings v2 features covered:

- Homepage
- Manage Identities
	- Phone
	- Username
	- Email
		- Change primary email
- Manage MFA
	- Phone (list)
	- Email (list)
	- TOTP (list)
	- Password (list + add)
	- Trusted devices
- Other security settings
	- Biometric
	- Password
	- Signed-in devices
- Schedule account deletion
	
---

ref DEV-2093
ref DEV-1840
ref DEV-2049
ref DEV-1928
ref DEV-1927
ref DEV-1973
ref DEV-1866
ref DEV-1926
ref DEV-1870
ref DEV-1923
ref DEV-1868
ref DEV-1929
ref DEV-1925
ref DEV-1922
ref DEV-1867
ref DEV-1963
ref DEV-1966
ref DEV-1960
ref DEV-1924
ref DEV-1921
ref DEV-1995
ref DEV-1994
ref DEV-1920
ref DEV-2043
ref DEV-1993
ref DEV-1992
ref DEV-2041
ref DEV-1969
ref DEV-1970
ref DEV-2009
ref DEV-2042
ref DEV-2045
ref DEV-2044
ref DEV-2015

### CODE REVIEW

File: .make-lint-translation-keys-expect [hunk 1]
```diff
@@ -25,7 +30,43 @@ resources/authgear/templates/en/web/authflowv2/forgot_password.html:103:30: temp
 resources/authgear/templates/en/web/authflowv2/forgot_password.html:159:26: template translation is forbidden: `__forgot_password_alternative`
 resources/authgear/templates/en/web/authflowv2/layout.html:5:14: template translation is forbidden: `widget`
 resources/authgear/templates/en/web/authflowv2/login.html:252:37: translation key not defined: "%s-icon"
+resources/authgear/templates/en/web/authflowv2/settings_biometric.html:33:24: translation key not defined: "__settings_biometric_item_remove_btn"
```
Comment (louischan-oursky) [inline .make-lint-translation-keys-expect:33]: Name these `__xxx.html`

File: authui/src/authflowv2/settingsDialog.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Controller } from "@hotwired/stimulus";
```
Comment (louischan-oursky) [inline authui/src/authflowv2/settingsDialog.ts:1]: This controller seems redundant.

File: authui/src/authflowv2/components/otp-input.css [hunk 3]
```diff
@@ -80,6 +80,7 @@
     @apply items-stretch;
     @apply gap-x-[var(--otp-input\_\_spacing)];
     @apply pointer-events-none;
+    @apply justify-center;
```
Comment (louischan-oursky) [inline authui/src/authflowv2/components/otp-input.css:83]: Revert this or you provide a reason for this change.

File: authui/src/authflowv2/base.css [hunk 4]
```diff
@@ -51,6 +51,10 @@
     /* --color-primary-theme-lighter: #d8e6fd; */
     /* --color-primary-theme-lighter-alt: #f5f9fe; */
 
+    /* secondary */
+    --color-secondary-theme-primary: #ff295b;
+    --color-secondary-theme-dark: #c4003d;
```
Comment (louischan-oursky) [inline authui/src/authflowv2/base.css:56]:
```
--color-destructive-primary: #ff295b;
--color-destructive-dark: #c4003d;
```

File: resources/authgear/templates/en/translation.json [hunk 5]
```diff
@@ -1115,26 +1119,117 @@
   "v2.page.select-account.default.description": "You have logged in with {email_is_present,select,true{{email}} other{{phone_number_is_present,select,true{{phone_number}} other{{preferred_username}}}}}. Proceed to continue.",
   "v2.page.select-account.default.title": "Log in to {AppOrClientName}",
   "v2.page.select-account.default.use-another-account": "Use another account",
+  "v2.page.settings-advanced-settings.default.title": "Advanced Settings",
+  "v2.page.settings-biometric.default.dialog-description": "Are you sure you want to disconnect biometric login for this device?",
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/translation.json:1123]: v2.page.settings-biometric.default.remove-biometric-dialog-description

File: resources/authgear/templates/en/translation.json [hunk 6]
```diff
@@ -1115,26 +1119,117 @@
...
+  "v2.page.settings-biometric.default.dialog-title": "Disconnect Biometric Login",
+  "v2.page.settings-biometric.default.disconnect-button-label": "Disconnect",
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/translation.json:1125]: remove-biometric-button-label

File: resources/authgear/templates/en/translation.json [hunk 7]
```diff
@@ -1115,26 +1119,117 @@
...
+  "v2.page.settings-biometric.default.item-description": "Created at <span data-date=\"{rfc3339}\" data-date-type=absolute data-date-date-style=short data-date-time-style=short>{time, datetime, short} UTC</span>",
+  "v2.page.settings-biometric.default.no-biometric-description": "You do not have any devices with biometrics enabled.",
+  "v2.page.settings-biometric.default.title": "Biometric Login",
+  "v2.page.settings-biometric.default.unknown-name-item-label": "Unknown Device",
+  "v2.page.settings-change-password.default.title": "Change Password",
+  "v2.page.settings-delete-account-success.default.button-label": "Done",
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/translation.json:1131]: done-button-label

File: resources/authgear/templates/en/translation.json [hunk 8]
```diff
@@ -1115,26 +1119,117 @@
...
+  "v2.page.settings-delete-account-success.default.description": "Your account has been deactivated and the data will be deleted on <span data-date=\"{rfc3339}\" data-date-type=absolute data-date-date-style=long data-date-time-style>{date, date, long}</span>",
+  "v2.page.settings-delete-account-success.default.title": "Account deletion scheduled",
+  "v2.page.settings-delete-account.default.button-label": "Delete Account",
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/translation.json:1134]: delete-account-button-label

File: resources/authgear/templates/en/translation.json [hunk 9]
```diff
@@ -1146,9 +1241,21 @@
...
-  "v2.page.settings-profile.default.navbar-title": "Profile",
   "v2.page.settings-profile.default.profile-picture-title": "Profile Picture",
+  "v2.page.settings-profile.default.title": "Profile",
   "v2.page.settings-profile.default.zoneinfo-title": "Timezone",
+  "v2.page.settings-sessions.default.all-sessions-dialog-description": "This will remove access to your account from all other devices",
+  "v2.page.settings-sessions.default.all-sessions-dialog-title": "Terminate all other sessions",
+  "v2.page.settings-sessions.default.country-ip-item-description": "{countryName} ({countryCode}) - {ip}",
+  "v2.page.settings-sessions.default.device-item-description": "{isCurrent, select, true {{desc} · <span class=\"settings-description--primary-color\">Active Now</span>} other {{desc} · <span data-date=\"{rfc3339}\">{time, datetime, short} UTC</span>}}",
+  "v2.page.settings-sessions.default.revoke-all-label": "Terminate all other sessions",
+  "v2.page.settings-sessions.default.session-dialog-description": "This will remove access to your account on this device",
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/translation.json:1252]: v2.page.settings-sessions.default.terminate-session-dialog-description

File: resources/authgear/templates/en/translation.json [hunk 10]
```diff
@@ -1146,9 +1241,21 @@
...
+  "v2.page.settings-sessions.default.session-dialog-title": "Terminate session",
+  "v2.page.settings-sessions.default.title": "Signed in Devices",
+  "v2.page.settings-totp.default.added-at-label": "Added at <span data-date=\"{rfc3339}\" data-date-type=absolute data-date-date-style=long>{time, datetime, short} UTC</span>",
+  "v2.page.settings-totp.default.button-add-authenticator-label": "+ Add authenticator app/device",
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/translation.json:1256]: add-authenticator-button-label

File: authui/src/authflowv2.ts [hunk 11]
```diff
@@ -96,6 +98,7 @@ Stimulus.register("image-picker", ImagePickerController);
 
 Stimulus.register("text-field", TextFieldController);
 Stimulus.register("dialog", DialogController);
+Stimulus.register("settings-dialog", SettingsDialogController);
```
Comment (louischan-oursky) [inline authui/src/authflowv2.ts:101]: Remove this.

File: pkg/auth/handler/webapp/authflowv2/routes.go [hunk 12]
```diff
@@ -71,13 +71,39 @@ const (
...
+
+	AuthflowV2RouteSettingsIdentityListPhone          = "/settings/identity/phone"
+	AuthflowV2RouteSettingsIdentityAddPhone           = "/settings/identity/add_phone"
+	AuthflowV2RouteSettingsIdentityEditPhone          = "/settings/identity/edit_phone"
+	AuthflowV2RouteSettingsIdentityViewPhone          = "/settings/identity/view_phone"
+	AuthflowV2RouteSettingsIdentityChangePrimaryPhone = "/settings/identity/change_primary_phone"
+	AuthflowV2RouteSettingsIdentityVerifyPhone        = "/settings/identity/verify_phone"
+
+	AuthflowV2RouteSettingsIdentityListUsername = "/settings/identity/username"
+	AuthflowV2RouteSettingsIdentityNewUsername  = "/settings/identity/new_username"
```
Comment (louischan-oursky) [inline pkg/auth/handler/webapp/authflowv2/routes.go:104]: See https://github.com/authgear/authgear-server/issues/3626 Verbs

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

File: authui/src/authflowv2/components/otp-input.css (hunk 3 above)
Comment (IniZio) [reply authui/src/authflowv2/components/otp-input.css]: Don't observe difference after disabling, removing this line first.

**Comment by IniZio** [comment]
@louischan-oursky Updated

File: resources/authgear/templates/en/translation.json [hunk 13]
```diff
@@ -110,7 +110,7 @@
   "copied-button-label": "Copied",
   "get-new-recovery-codes-button-label": "Get new codes",
   "download-button-label": "Download",
-  "disconnect-button-label": "Disconnect",
+  "remove-biometric-button-label": "Disconnect",
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/translation.json:113]: Do not change v1 translation key.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

File: resources/authgear/templates/en/translation.json (hunk 13 above)
Comment (IniZio) [reply resources/authgear/templates/en/translation.json]: This commit was actually fixed up last week, just tried rebase and pushed again

**Comment by louischan-oursky** [comment]
@IniZio Still failed.

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**